internal/tfcontext/context.go

```diff
@@ -0,0 +1,23 @@
+package tfcontext
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+const endpointContextKey = "endpoint"
+const tokenContextKey = "token"
+
+const methodContextKey = "method"
+const apiPathContextKey = "path"
+
+func AddProviderInfo(ctx context.Context, endpoint, token string) context.Context {
+	ctx = tflog.SetField(ctx, endpointContextKey, endpoint)
+	return tflog.SetField(ctx, tokenContextKey, token)
```

Comment:
Maybe let's not log the token? It's sensitive and better not be in the logs.

Reply:
removed